webdriver/tests/bidi/storage/get_cookies/partition.py

```diff
@@ -0,0 +1,230 @@
+        partition=BrowsingContextPartitionDescriptor(top_context["context"])
+    )
+
+    assert cookies == {
+        "cookies": [],
+        "partitionKey": {"sourceOrigin": source_origin_1},
+    }
+
+    await remove_cookie(bidi_session, new_tab["context"], cookie_name)
+
+
+@pytest.mark.parametrize("domain", ["", "alt"], ids=["same_origin", "cross_origin"])
+async def test_partition_context_iframe(
+    bidi_session, new_tab, inline, domain_value, origin, domain
+):
+    iframe_url = inline("<div id='in-iframe'>foo</div>", domain=domain)
+    page_url = inline(f"<iframe src='{iframe_url}'></iframe>")
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=page_url, wait="complete"
+    )
+
+    contexts = await bidi_session.browsing_context.get_tree(root=new_tab["context"])
+    iframe_context = contexts[0]["children"][0]
+
+    cookie_name = "foo"
+    cookie_value = "bar"
+    await add_cookie(bidi_session, iframe_context["context"], cookie_name, cookie_value)
+
+    # Check that added cookies are present on the right context
+    cookies = await bidi_session.storage.get_cookies(
+        partition=BrowsingContextPartitionDescriptor(iframe_context["context"])
+    )
+
+    assert cookies["partitionKey"] == {
+        "sourceOrigin": origin(protocol="https", domain=domain)
+    }
+    assert len(cookies["cookies"]) == 1
+    recursive_compare(
+        {
+            "domain": domain_value(domain=domain),
+            "httpOnly": False,
+            "name": cookie_name,
+            "path": "/webdriver/tests/support",
+            "sameSite": "none",
+            "secure": False,
+            "size": 6,
+            "value": {"type": "string", "value": cookie_value},
+        },
+        cookies["cookies"][0],
+    )
+
+    await remove_cookie(bidi_session, iframe_context["context"], cookie_name)
+
+
+@pytest.mark.parametrize(
+    "protocol",
+    [
+        "http",
+        "https",
+    ],
+)
+async def test_partition_source_origin(
+    bidi_session, top_context, new_tab, inline, domain_value, origin, protocol
+):
+    url_1 = inline("<div>foo</div>", domain="alt", protocol=protocol)
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url_1, wait="complete"
+    )
+    source_origin_1 = origin(protocol, "alt")
+
+    url_2 = inline("<div>bar</div>", protocol=protocol)
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=url_2, wait="complete"
+    )
+    source_origin_2 = origin(protocol)
+
+    cookie_name = "foo"
+    cookie_value = "bar"
+    await add_cookie(bidi_session, new_tab["context"], cookie_name, cookie_value)
+
+    # Check that added cookies are present on the right context
+    cookies = await bidi_session.storage.get_cookies(
+        partition=StorageKeyPartitionDescriptor(source_origin=url_2)
+    )
+
+    assert cookies["partitionKey"] == {"sourceOrigin": source_origin_2}
```

Comment:
For what I understand, the `sourceOrigin` should be present only in case of **partitioned** cookie. Do I misunderstand it? @lutien

Reply:
This part is not 100% clear to me, so I might be wrong, but from what I understood from the spec, [this comment](https://github.com/w3c/webdriver-bidi/pull/501#issuecomment-1725180497) and [this comment](https://github.com/w3c/webdriver-bidi/pull/501#discussion_r1378725423) from James, the idea is to return as `partitionKey` the key which was used to retrieve the cookies. So it's not the key of the cookies. 
cc @jgraham (in case I'm completely wrong)

Reply:
Oh, agree. The `partitionKey` should be there, as the cookies were requested with it.

Reply:
Or disagree, depending on definition of 
> browsing context’s [associated storage partition](https://w3c.github.io/webdriver-bidi/#associated-storage-partition).

Which references
> Each [browsing context](https://html.spec.whatwg.org/multipage/document-sequences.html#browsing-context) also has an associated storage partition, which is the [storage partition](https://w3c.github.io/webdriver-bidi/#storage-partition) it uses to persist data.

So the answer depends on that if browsing context's "associated storage partition" contains "source origin" or not.